game-core/src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -2313,14 +2342,18 @@ private Matches() {}
       final GamePlayer player,
       final int battleRound,
       final boolean includeAttackersThatCanNotMove,
-      final boolean doNotIncludeBombardingSeaUnits) {
+      final boolean doNotIncludeBombardingSeaUnits,
+      final Collection<UnitType> firingUnits) {
 
-    // Filter out anything like factories, or units that have no combat ability AND cannot be taken
-    // casualty
     final PredicateBuilder<UnitType> canBeInBattleBuilder =
+        // remove infrastructure units
         PredicateBuilder.of(unitTypeIsInfrastructure().negate())
+            // unless it can support or fight
             .or(unitTypeIsSupporterOrHasCombatAbility(attack, player))
```

Comment:
Intermediate variables can perhaps help reduce the commentary. For example:
```
var isAaThatCanFireThisRound = unitType -> unitTypeIsAaForCombatOnly().and(unitTypeIsAaThatCanFireOnRound(battleRound);
```

WDYT of using that technique to avoid the inline commentary? I think that is only needed for the more complex conditions, the simpler ones, the code might be able to just speak for itself.

Reply:
For a long series of statements, having one comment to describe the general outcome and what is generally happening can be a good thing. Make it easier for a reader to know what to expect and makes it easier to see how each statement fits in to the overall outcome. Commenting each line though is more questionable and can speak to inarticulate code (which can be fixed by improving naming, intermediate variables, private helper methods, re-organizing methods).

Reply:
I personally don't see the intermediate variables as being helpful in explaining what is happening.  I'll just move all the comments to above the definition of the `canBeInBattleBuilder` variable.

Reply:
Fair enough. When it comes to "what is the code doing" commentary, reducing that can help avoid coupling (meaning the code can be changed and the comments do not become out of date) and sometimes such commentary is being used as a bandaid for inexpressive code. There is a right balance though, some "what is the code doing"  commentary can really help set expectations of what a person is about to read and makes it easier to understand (easier to understand => faster to read => quicker to modify => and more comprehensive leads to fewer bugs)